bigframes/operations/_matplotlib/hist.py

```diff
@@ -0,0 +1,169 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+from typing import Literal
+
+import numpy as np
+import pandas as pd
+
+import bigframes.constants as constants
+import bigframes.operations._matplotlib.core as bfplt
+
+
+class HistPlot(bfplt.MPLPlot):
+    @property
+    def _kind(self) -> Literal["hist"]:
+        return "hist"
+
+    def __init__(
+        self,
+        data,
+        bins: int = 10,
+        **kwargs,
+    ) -> None:
+        self.bins = bins
+        self.label = kwargs.get("label", None)
+        self.by = kwargs.pop("by", None)
+        self.kwargs = kwargs
+
+        if self.by is not None:
+            raise NotImplementedError(
+                f"Non-none `by` argument is not yet supported. {constants.FEEDBACK_LINK}"
+            )
+        if not isinstance(self.bins, int):
+            raise NotImplementedError(
+                f"Only integer values are supported for the `bins` argument. {constants.FEEDBACK_LINK}"
+            )
+        if kwargs.get("weight", None) is not None:
+            raise NotImplementedError(
+                f"Non-none `weight` argument is not yet supported. {constants.FEEDBACK_LINK}"
+            )
+
+        self.data = self._compute_plot_data(data)
+
+    def generate(self) -> None:
+        """
+        Calculates weighted histograms through BigQuery and plots them through pandas
+        native histogram plot.
+        """
+        hist_bars = self._calculate_hist_bars(self.data, self.bins)
+        bin_edges = self._calculate_bin_edges(
+            hist_bars, self.bins, self.kwargs.get("range", None)
+        )
+
+        weights = {
+            col_name: hist_bar.values for col_name, hist_bar in hist_bars.items()
+        }
+        hist_x = {
+            col_name: pd.Series(
+                (
+                    hist_bar.index.get_level_values("left_exclusive")
+                    + hist_bar.index.get_level_values("right_inclusive")
+                )
+                / 2.0
+            )
+            for col_name, hist_bar in hist_bars.items()
+        }
+
+        # Align DataFrames for plotting despite potential differences in column
+        # lengths, filling shorter columns with zeros.
+        hist_x_pd = pd.DataFrame(
+            list(itertools.zip_longest(*hist_x.values())), columns=list(hist_x.keys())
+        ).sort_index(axis=1)[self.data.columns.values]
+        weights_pd = pd.DataFrame(
+            list(itertools.zip_longest(*weights.values())), columns=list(weights.keys())
+        ).sort_index(axis=1)[self.data.columns.values]
+        hist_x_pd.fillna(0, inplace=True)
+        weights_pd.fillna(0, inplace=True)
+
+        self.axes = hist_x_pd.plot.hist(
+            bins=bin_edges,
+            weights=np.array(weights_pd.values),
```

Comment:
Is this method still not working on pandas 1.5 or did the fillna solve that?

Reply:
the fillna solves the issue in pandas 1.5. I can add a comment above calling.